e2e/testcafe-devextreme/tests/editors/checkBox/common.ts

```diff
@@ -1,49 +1,79 @@
+const READONLY_STATE_CLASS = 'dx-state-readonly';
+const DEFAULT_STATE_CLASS = '';
+const ACTIVE_STATE_CLASS = 'dx-state-active';
+const HOVER_STATE_CLASS = 'dx-state-hover';
+const FOCUSED_STATE_CLASS = 'dx-state-focused';
+const DISABLED_STATE_CLASS = 'dx-state-disabled';
 
-fixture.disablePageReloads`CheckBox`
+fixture.disablePageReloads`CheckBox render`
   .page(url(__dirname, '../../container.html'));
 
-[false, true].forEach((isColumnCountStyle) => {
-  test(`Render ${!isColumnCountStyle ? 'default' : 'with column-count style on container'}`, async (t) => {
-    const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+[false, true].forEach((rtlEnabled) => {
+  [20, 30].forEach((iconSize) => {
+    [false, true].forEach((limitedWidth) => {
+      test(`Checkbox styles, rtlEnabled=${rtlEnabled} iconSize=${iconSize} limitedWidth=${limitedWidth}`, async (t) => {
+        const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+        await testScreenshot(t, takeScreenshot, `CheckBox states rtlEnabled=${rtlEnabled} iconSize=${iconSize} limitedWidth=${limitedWidth}.png`, { element: '#container', shouldTestInCompact: true });
 
-    await testScreenshot(t, takeScreenshot, `Checkbox states${isColumnCountStyle ? ' with column count style' : ''}.png`, { element: '#container', shouldTestInCompact: true });
+        await testScreenshot(t, takeScreenshot, `CheckBox states rtlEnabled=${rtlEnabled} iconSize=${iconSize} limitedWidth=${limitedWidth}.png`, { element: '#container', theme: getFullThemeName().replace('light', 'dark') });
 
-    await t
-      .expect(compareResults.isValid())
-      .ok(compareResults.errorMessages());
-  }).before(async () => {
-    await setStyleAttribute(Selector('#container'), `padding: 5px; width: 300px; height: 200px; ${isColumnCountStyle ? 'column-count: 2' : ''}`);
+        await t
+          .expect(compareResults.isValid())
+          .ok(compareResults.errorMessages());
+      }).before(async (t) => {
+        t.ctx.ids = [];
 
-    await insertStylesheetRulesToPage(`.${CHECKBOX_CLASS} { display: block; }`);
+        const croppedWidth = iconSize === 20 ? 400 : 450;
+        const fullWidth = iconSize === 20 ? 450 : 500;
 
-    await appendElementTo('#container', 'div', 'checked');
-    await createWidget('dxCheckBox', { value: true, text: 'checked' }, '#checked');
+        await setStyleAttribute(Selector('#container'), `padding: 5px; width: ${limitedWidth ? croppedWidth : fullWidth}px; height: 200px; column-count: 6;`);
 
-    await appendElementTo('#container', 'div', 'unchecked');
-    await createWidget('dxCheckBox', { value: false, text: 'unchecked' }, '#unchecked');
+        await insertStylesheetRulesToPage(`.${CHECKBOX_CLASS} { display: block; }`);
 
-    await appendElementTo('#container', 'div', 'indeterminate');
-    await createWidget('dxCheckBox', { value: undefined, text: 'indeterminate' }, '#indeterminate');
+        for (const state of [
```

Comment:
Is there a possibility to reduce the number of screenshots and test more configurations on one screenshot?